src/Microsoft.Framework.FileSystemGlobbing/Abstractions/DirectoryInfoWrapper.cs

```diff
@@ -9,31 +10,48 @@ namespace Microsoft.Framework.FileSystemGlobbing.Abstractions
     public class DirectoryInfoWrapper : DirectoryInfoBase
     {
         private readonly DirectoryInfo _directoryInfo;
+        private readonly bool _isParentPath;
 
-        public DirectoryInfoWrapper(DirectoryInfo directoryInfo)
+        public DirectoryInfoWrapper(DirectoryInfo directoryInfo, bool isParentPath = false)
         {
             _directoryInfo = directoryInfo;
+            _isParentPath = isParentPath;
         }
 
         public override IEnumerable<FileSystemInfoBase> EnumerateFileSystemInfos(string searchPattern, SearchOption searchOption)
         {
-            foreach (var fileSystemInfo in _directoryInfo.EnumerateFileSystemInfos(searchPattern, searchOption))
+            if (_directoryInfo.Exists)
             {
-                var directoryInfo = fileSystemInfo as DirectoryInfo;
-                if (directoryInfo != null)
+                foreach (var fileSystemInfo in _directoryInfo.EnumerateFileSystemInfos(searchPattern, searchOption))
                 {
-                    yield return new DirectoryInfoWrapper(directoryInfo);
-                }
-                else
-                {
-                    yield return new FileInfoWrapper((FileInfo)fileSystemInfo);
+                    var directoryInfo = fileSystemInfo as DirectoryInfo;
+                    if (directoryInfo != null)
+                    {
+                        yield return new DirectoryInfoWrapper(directoryInfo);
```

Comment:
Since the directories returned from EnumerateFIleSystemInfo are always legit folder so they won't be parent directory. Do you think call it out as `false` could make the code more readable?